Add Dataset(Extent) constructor to Python bindings

franzpoeschel opened the pull request:
There's no Python-based test for resizing since resizing did not add any new API, but was instead a backend-only (~middleend rather) thing. If we want to test this, I suggest rebasing it onto https://github.com/openPMD/openPMD-api/pull/1060?
Otherwise, this just adds a small missing constructor to the Python bindings.

franzpoeschel committed:
Add Dataset(Extent) constructor to Python bindings

ax3l commented:
Thank you! When we rebase this, let's add the doc string here as well:
https://github.com/openPMD/openPMD-api/pull/1060#issuecomment-885555127 :tada:

ax3l committed:
Resize Example: Easier Dataset

This example works easier with nested logic, showing that one
does not need to carry around the original type and options
for changing the extent later on.

ax3l commented on examples/3b_write_resizable_particles.py:
```diff
@@ -47,7 +47,7 @@
     y = np.array([-7., -8., -9.], dtype=np.double)
     offset += dataset.extent[0]
-    dataset.extend([dataset.extent[0] + x.shape[0]])
+    dataset = io.Dataset([dataset.extent[0] + x.shape[0]])
```
In line 55 on macOS:
```
  File "/Users/runner/work/openPMD-api/openPMD-api/build/bin/3b_write_resizable_particles.py", line 55, in <module>
    rc_x[offset:] = x
RuntimeError: Datatypes of chunk data (DOUBLE) and record component (UNDEFINED) do not match.
[HDF5] Datatype::UNDEFINED caught during dataset creation (serial HDF5)
```

ax3l replied:
Ah, I forgot to flush after the first write :)
